src/main/java/br/dev/marcoalmeida/mb/service/MovieService.java

```diff
@@ -31,95 +32,142 @@
-        StatefulBeanToCsv<MovieDTO> statefulBeanToCsv = new StatefulBeanToCsvBuilder<MovieDTO>(writer)
-                .withSeparator(CSVWriter.DEFAULT_SEPARATOR)
-                .build();
-
-        statefulBeanToCsv.write(generateByTitle(title).stream()
-                .map(MovieMapper.INSTANCE::convert)
-                .toList());
-
-    }
-
-    public Optional<ReflexivePair<Movie>> findNewPair(Set<ReflexivePair<Movie>> usedPairs){
-        long retryCount = 0;
-
-        while (retryCount < MAX_RETRIES){
-            Optional<ReflexivePair<Movie>> candidate = selectRandomReflexivePair();
-
-            if (candidate.stream().anyMatch(pair -> !usedPairs.contains(pair))){
-                return candidate;
-            }
-
-            retryCount++;
-        }
-
-
-        return Optional.empty();
-
-    }
-
-    private Optional<ReflexivePair<Movie>> selectRandomReflexivePair() {
-        return selectRandomMovie(Set.of())
-                .flatMap(this::findAnotherMovie);
-    }
-
-    private Optional<ReflexivePair<Movie>> findAnotherMovie(Movie m1){
-       return selectRandomMovie(Set.of(m1.getId()))
-                .flatMap(m2 -> Optional.of(new ReflexivePair<Movie>(m1, m2)));
-    }
-
-    private Optional<Movie> selectRandomMovie(Set<String> setMovieId) {
-        return movieRepository.findByIdNotIn(setMovieId, PageRequest.of(RandomUtils.randomNaturalNumberUpTo(
-                                movieRepository.countByIdNotIn(setMovieId)), 1))
-                .get()
-                .findFirst();
-    }
-    private List<Movie> saveResults(ResultsDTO resultsDTO) {
-        log.info("resultsDTO123131 [{}]", resultsDTO);
-        return resultsDTO.getResults().stream()
-                .flatMap(resultItem -> getAdditionalInfo(resultItem).stream())
-                .collect(Collectors.toList());
-    }
-
-    private Optional<Movie> getAdditionalInfo(SearchResultDTO resultItem) {
-        return Optional.ofNullable(omdbClient.getInfo(resultItem.getImdbID()).getBody())
-                .map(info -> save(resultItem, info));
-    }
-
-
-    private Movie save(SearchResultDTO resultItem, InfoDTO info) {
-        return movieRepository.save(Movie.builder()
-                .id(resultItem.getImdbID())
-                .title(resultItem.getTitle())
-                .rating(info.getImdbRating())
-                .votes(Long.parseLong(info.getImdbVotes().replace(",", "")))
-                .posterUrl(resultItem.getPoster())
-                .releaseYear(getReleaseYear(info))
-
-                .build());
-    }
-
-    private static Long getReleaseYear(InfoDTO info) {
-        try {
-            return Long.parseLong(info.getYear());
-        } catch (NumberFormatException e) {
-            return null;
-        }
-    }
-
+	private final OmdbClient omdbClient;
+
+	private final MovieRepository movieRepository;
+
+	private static final Long MAX_RETRIES = 1000L;
+	
```

Comment:
start your analysis from here, and ignore attributes and methods related to findNewPair()

This is a good example of a class that needs to be broken into two.

Reply:
Breaking the class into two -> great exercise in refactoring